allen-cell-animated/simularium-website
# Add tests for ResultsPanel selectors (100% coverage for all selectors)

Problem
=======
Resolves #182 

This is the last PR for the above issue! After this is merged, all selectors in simularium-website will be 100% covered.

Solution
========
Added tests to the `ResultsPanel` container's selector and its helper functions, breaking down the functions further to make them more easily testable on the way.

## Type of change
Please delete options that are not relevant.

* This change requires updated or new tests

Steps to Verify:
----------------
1. Check that refactoring the selector and helper functions didn't break the plots. Run `npm start`, load some models with plots (like endocytosis) and make sure the plots look formatted the same way as before.
2. GitHub Actions should take care of test coverage check

Keyfiles (delete if not relevant):
-----------------------
1. src/containers/ResultsPanel/selectors.test.ts - All new tests
2. src/containers/ResultsPanel/selectors.ts - Broke out `configureLayout` helper function into smaller functions to make them more easily testable; turned some style values into constants
